### PR TITLE
User Registration

### PR DESCRIPTION
**Pull request by danieloluwadare**
All left is to generate TestCases

### CODE REVIEW

File: pkg/config/mail/mailgunImpl.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+import (
+	"context"
+	"fmt"
+	"github.com/joho/godotenv"
+	"github.com/mailgun/mailgun-go/v3"
+	"os"
+	"time"
+)
+//func INewService(repository irepository.IUserRepository) iservice.IUserService {
```
Comment (mr-m0nkey) [inline pkg/config/mail/mailgunImpl.go]: Is this code no longer needed?

Comment (danieloluwadare) [reply pkg/config/mail/mailgunImpl.go]: Will fix it soon

Comment (danieloluwadare) [reply pkg/config/mail/mailgunImpl.go]: its not actually necessary

File: pkg/config/mail/mailgunImpl.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+
+type mailGunImplementation struct {
+	domain string
+	apiKey string
+
+}
+
+
+func NewMailGunImplementationNoArgs() IMail {
```
Comment (mr-m0nkey) [inline pkg/config/mail/mailgunImpl.go:17]: I'm not sure if there's method overloading in go but if there is then `NewMailGunImplementation` and `NewMailGunImplementationNoArgs ` can have the same name but take different parameters

Comment (danieloluwadare) [reply pkg/config/mail/mailgunImpl.go:17]: actually no method overloading.. actually NewMailGunImplementation for the purpose of testing and if you wanted to inject a different parameter
NewMailGunImplementationNoArgs=>to be used on the fly

**Review by mr-m0nkey — COMMENTED** [review]
Please can you also run `go fmt` so it can apply the go standard formatting to it

File: pkg/config/mail/mailgunImpl.go (hunk 2 above)
Comment (mr-m0nkey) [reply pkg/config/mail/mailgunImpl.go:17]: Okay, thanks

**Review by mr-m0nkey — APPROVED** [review]

**Pull request merged**